phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -859,39 +899,54 @@ public boolean verifySingleIndexRow(Result indexRow, IndexToolVerificationResult
                         (actual instanceof Delete && isDeleteFamily(actual))) {
                     expectedIndex++;
                     actualIndex++;
-                    matchingCount++;
                     continue;
                 }
             }
-            if (matchingCount > 0) {
-                break;
+            break;
+        }
+
+        if (expectedIndex == expectedSize ){
+            // every expected mutation has its matching one in the actual list.
+            verificationPhaseResult.validIndexRowCount++;
+            return true;
+        }
+
+        if (isTimestampBeyondMaxLookBack(currentTime, getTimestamp(expectedMutationList.get(expectedIndex)))){
+            if (expectedIndex > 0) {
+                // if current expected index mutation is beyond max look back window, we only need to make sure its latest
+                // mutation is a matching one, as an SCN query is required.
```

Comment:
per the logline, should we expect a scan here?

Reply:
which log line are you talking about? The comment says it is SCN query.